### DISCUSSION TITLE
Access denied using proxy

### DISCUSSION THREAD

**Discussion by q1165600785** [earthaccess-dev/earthaccess#805] [Q&A] [answered]
I am in China and I have to use a proxy. Is there any good solution for this?

requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://data.gesdisc.earthdata.nasa.gov/data/GLDAS/GLDAS_NOAH025_3H.2.1/2018/001/GLDAS_NOAH025_3H.A20180101.0000.021.nc4
When I copy the link into the Chrome browser, I can download it.

**Comment by chuckwondo** [comment]
You should be able to set the `HTTPS_PROXY` environment value to the proxy address (e.g., http://host:port). Your browser likely has the proxy configured, which is likely why you can download via the browser.

**Reply by mfisher87** [reply]
I moved this discussion to the Q&A category - please select "Mark as answer" if this comment resolves your issue! :point_up:

**Reply by q1165600785** [reply]
Is there a method to set up a proxy for EarthAccess

**Reply by chuckwondo** [reply]
Please try setting your `HTTPS_PROXY` environment variable as mentioned above, and let us know if that works for you.

**Reply by q1165600785** [reply]
I really feel the big shot's answer. I used another Python package called GEEMAP, which has a method to directly set the proxy, GEEMAP. set_dexy (port=8880), which is very convenient. I think Earthacess should also define a similar method globally to directly set the proxy without unnecessary trouble.

**Comment by chuckwondo** [comment, marked as answer]
@q1165600785, please see this workaround: https://github.com/nsidc/earthaccess/issues/501#issuecomment-2377914600

**Reply by q1165600785** [reply]
I really feel the big shot's answer. I used another Python package called GEEMAP, which has a method to directly set the proxy, GEEMAP. set_dexy (port=8880), which is very convenient. I think Earthacess should also define a similar method globally to directly set the proxy without unnecessary trouble.
